includes/class-indieauth-token-endpoint.php

```diff
@@ -151,84 +161,64 @@ public function request( $params ) {
+				$user = get_user_by_identifier( $response['me'] );
+				$user = $user->ID;
 			}
-			if ( $token ) {
-				// Return only the standard keys in the response
-				return new WP_REST_Response(
-					wp_array_slice_assoc(
-						$token,
-						array(
-							'access_token',
-							'token_type',
-							'scope',
-							'me',
-							'profile',
-						)
-					),
-					200, // Status Code
-					array(
-						'Cache-Control' => 'no-store',
-						'Pragma'        => 'no-cache',
-					)
-				);
+			if ( in_array( 'profile', $scopes, true ) ) {
+				$token['profile'] = indieauth_get_user( $user, in_array( 'email', $scopes, true ) );
 			}
-		} else {
-			return new WP_OAuth_Response( 'invalid_grant', __( 'This authorization code was issued with no scope, so it cannot be used to obtain an access token', 'indieauth' ), 400 );
-		}
-		return new WP_OAuth_Response( 'server_error', __( 'There was an error issuing the access token', 'indieauth' ), 500 );
-	}
 
-	public static function verify_local_authorization_code( $post_args ) {
-		$tokens = new Token_User( '_indieauth_code_' );
-		$return = $tokens->get( $post_args['code'] );
-		if ( ! $return ) {
-			return new WP_OAuth_Response( 'invalid_code', __( 'Invalid authorization code', 'indieauth' ), 401 );
-		}
-		if ( isset( $return['code_challenge'] ) ) {
-			if ( ! isset( $post_args['code_verifier'] ) ) {
-				$tokens->destroy( $post_args['code'] );
-				return new WP_OAuth_Response( 'invalid_grant', __( 'Failed PKCE Validation', 'indieauth' ), 400 );
-			}
-			if ( ! pkce_verifier( $return['code_challenge'], $post_args['code_verifier'], $return['code_challenge_method'] ) ) {
-				$tokens->destroy( $post_args['code'] );
-				return new WP_OAuth_Response( 'invalid_grant', __( 'Failed PKCE Validation', 'indieauth' ), 400 );
+			// Issue a token
+			if ( ! empty( array_diff( $scopes, array( 'profile', 'email' ) ) ) && array( 'profile' ) !== $scopes ) {
+				$info                  = new IndieAuth_Client_Discovery( $params['client_id'] );
+				$return['token_type']  = 'Bearer';
+				$return['scope']       = $response['scope'];
+				$return['issued_by']   = rest_url( 'indieauth/1.0/token' );
+				$return['client_id']   = $params['client_id'];
+				$return['client_name'] = $info->get_name();
+				$return['client_icon'] = $info->get_icon();
+				$return['issued_at']   = time();
+				$return                = array_filter( $return );
+
+				$return['access_token'] = $this->set_token( $return );
 			}
-			unset( $return['code_challenge'] );
-			unset( $return['code_challenge_method'] );
 		}
-		$return['me'] = get_url_from_user( $return['user'] );
 
-		$user = get_user_by( 'id', $return['user'] );
-		if ( $user ) {
-			$return['profile'] = indieauth_get_user( $user );
+		if ( $return ) {
+			// Return only the standard keys in the response
+			return new WP_REST_Response(
+				wp_array_slice_assoc(
+					$return,
+					array(
+						'access_token',
+						'token_type',
+						'scope',
+						'me',
+						'profile',
```

Comment:
Maybe I am missing something, but when is `profile` added to `$return`? Line 182 adds it to `$token`. And with the removal of the old line 227, I am not sure when it is being added at all.

Reply:
Fixing that one.